source/val/validate_adjacency.cpp

```diff
@@ -27,25 +27,39 @@
 namespace spvtools {
 namespace val {
 
+spv_result_t ValidateAdjacencyForOpPhi(ValidationState_t& _) {
+  const auto& instructions = _.ordered_instructions();
+  bool valid_to_place = false;
+
+  for (size_t i = 0; i < instructions.size(); ++i) {
+    const auto& inst = instructions[i];
+    switch (inst.opcode()) {
+      case SpvOpLabel:
+        valid_to_place = true;
+        break;
+      case SpvOpPhi:
+        if (!valid_to_place) {
```

Comment:
Can you add a test for the following:

```
...
%func = OpFunction ...
%entry = OpLabel
%1 = OpPhi %int
OpReturn
OpFunctionEnd
...
```

This should be invalid, however, I'm not sure if we check it

Reply:
This is checked [here](https://github.com/KhronosGroup/SPIRV-Tools/blob/2c2fee797908ffb1aab96a3b7265baaada1a0e12/test/val/val_id_test.cpp#L4321)

Reply:
Oh sorry. I was confused. Is the problem that `OpPhi` is located in the entry block of the function?

Reply:
Yes, it will have zero predecessors in that case which would seem to match.

Reply:
Done.